sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5644,6 +5672,83 @@ pi_result piTearDown(void *PluginParameter) {
   delete PiPlatformsCache;
   delete PiPlatformsCacheMutex;
 
+  // Print the balance of various create/destroy native calls.
+  if (ZeDebug & ZE_DEBUG_CALL_COUNT) {
+    // clang-format off
+    //
+    // The format of this table is such that each row accounts for a
+    // specific type of objects, and all elements in the raw except the last
+    // one are allocating objects of that type, while the last element is known
+    // to deallocate objects of that type.
+    //
+    std::vector<std::vector<std::string>> CreateDestroySet = {
+      {"zeContextCreate",      "zeContextDestroy"},
+      {"zeCommandQueueCreate", "zeCommandQueueDestroy"},
+      {"zeModuleCreate",       "zeModuleDestroy"},
+      {"zeKernelCreate",       "zeKernelDestroy"},
+      {"zeEventPoolCreate",    "zeEventPoolDestroy"},
+      {"zeCommandListCreateImmediate", "zeCommandListCreate", "zeCommandListDestroy"},
+      {"zeEventCreate",        "zeEventDestroy"},
+      {"zeFenceCreate",        "zeFenceDestroy"},
+      {"zeImageCreate",        "zeImageDestroy"},
+      {"zeSamplerCreate",      "zeSamplerDestroy"},
+      {"zeMemAllocDevice", "zeMemAllocHost", "zeMemAllocShared", "zeMemFree"},
+    };
```

Comment:
Do you also want to match zeDeviceGet() and zeDeviceDestroy()?

Reply:
`zeDeviceGet` returns multiple devices (and they are not allocated but returned from a cache) and there is no `zeDeviceDestroy` (the devices' cache is destroyed at the end)